erts/preloaded/src/prim_inet.erl

```diff
@@ -1884,6 +1884,8 @@ type_value_2(uint8, X)  when X band 16#ff =:= X       -> true;
 type_value_2(time, infinity)                          -> true;
 type_value_2(time, X) when is_integer(X), X >= 0      -> true;
 type_value_2(ip,{A,B,C,D}) when ?ip(A,B,C,D)          -> true;
+type_value_2(ip,{A,B,C,D,E,F,G,H})
+  when ?ip6(A,B,C,D,E,F,G,H)                          -> true;
```

Comment:
`ip` today means IPv4 address.  It should not be extended to allow IPv6 address.
(Maybe it should be renamed to `ip4` then we could use `ip` for `ip4` | `ip6`)

in `enc_value_3` a type `ip` value is encoded from `{A,B,C,D}` to a IPv4 binary address.  Your change allows to pass an IPv6 address to this point that fails this encoding.

There is also no corresponding code in `inet_drv.c` that handles an IPv6 address, so that would probably segfault if you send down an encoded IPv6 address.  This missing corresponding code in `inet_drv.c` should look at the socket type to figure which address to expect, whether to use IPV6_ADD_MEMBERSHIP or IP_ADD_MEMBERSHIP, and recognize when it gets an address on the wrong format.